api/audit/models.py

```diff
@@ -72,17 +91,13 @@ def get_history_record_model_class(
         module = import_module(module_path)
         return getattr(module, class_name)
 
-    @hook(AFTER_SAVE)
+    @hook(
+        AFTER_CREATE,
+        priority=priority.HIGHEST_PRIORITY,
+        when="environment_document_updated",
```

Comment:
Ah, I didn't realise you could use properties here, that's quite cool.

Reply:
Yeah, it's pretty cool